src/openloco/windows/CompanyFaceSelection.cpp

```diff
@@ -1,15 +1,320 @@
+#include "../audio/audio.h"
+#include "../companymgr.h"
+#include "../core/Optional.hpp"
+#include "../game_commands.h"
+#include "../graphics/colours.h"
+#include "../graphics/image_ids.h"
+#include "../input.h"
 #include "../interop/interop.hpp"
+#include "../localisation/FormatArguments.hpp"
+#include "../objects/competitor_object.h"
+#include "../objects/interface_skin_object.h"
+#include "../objects/objectmgr.h"
 #include "../openloco.h"
 #include "../ui/WindowManager.h"
 
 using namespace openloco::interop;
 
 namespace openloco::ui::windows::CompanyFaceSelection
 {
-    void open(company_id_t id)
+    static loco_global<company_id_t, 0x9C68F2> _9C68F2; // Use in a game command??
+    static loco_global<uint16_t, 0x112C1C1> _numberCompetitorObjects;
+    static loco_global<int32_t, 0x112C876> _currentFontSpriteBase;
+    static loco_global<competitor_object*, 0x0050D15C> _loadedObject; // This could be any type of object
+    static loco_global<int32_t, 0x0113E72C> _cursorX;
+    static loco_global<string_id, 0x009C68E8> gGameCommandErrorTitle;
+
+    static const gfx::ui_size_t windowSize = { 400, 272 };
+    static constexpr uint32_t rowHeight = 10;
+    static window_event_list events;
+
+    enum widx
+    {
+        frame,
+        caption,
+        close_button,
+        panel,
+        scrollview,
+        face_frame
+    };
+
+    // 0x509680
+    static widget_t widgets[] = {
+        make_widget({ 0, 0 }, windowSize, widget_type::frame, 0),
+        make_widget({ 1, 1 }, { 398, 13 }, widget_type::caption_24, 0, string_ids::company_face_selection_title),
+        make_widget({ 385, 2 }, { 13, 13 }, widget_type::wt_9, 0, image_ids::close_button, string_ids::tooltip_close_window),
+        make_widget({ 0, 15 }, { 400, 257 }, widget_type::panel, 1),
+        make_widget({ 4, 19 }, { 188, 248 }, widget_type::scrollview, 1, vertical, string_ids::tooltip_company_face_selection),
+        make_widget({ 265, 23 }, { 66, 66 }, widget_type::wt_5, 1),
+        widget_end(),
+    };
+
+    static void initEvents();
+
+    static std::vector<uint32_t> _inUseCompetitors;
+
+    // Object free?
+    static void sub_471B95()
+    {
+        call(0x00471B95);
+    }
+
+    // Object load?
+    static void sub_47176D(objectmgr::header& object)
+    {
+        registers regs;
+        regs.ebp = reinterpret_cast<int32_t>(&object);
+        call(0x0047176D, regs);
+    }
+
+    // 0x004720EB Returns std::nullopt if not loaded
```

Comment:
To prevent this not matching regexes looking for just an offset comment, I suggest this is split:
```suggestion
    // Returns std::nullopt if not loaded
    // 0x004720EB
```